godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/PoolByteArray.kt

```diff
@@ -0,0 +1,166 @@
+     */
+    fun appendArray(array: PoolByteArray) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_append_array(it, array.handle.ptr))
+        }
+    }
+
+    /**
+     * Returns true if the array is empty.
+     */
+    fun empty() {
+        callNative {
+            checkNotNull(Godot.gdnative12.godot_pool_byte_array_empty(it))
+        }
+    }
+
+    /**
+     *  Retrieve the element at the given index.
+     */
+    operator fun get(idx: Int): UByte {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_get(it, idx))
+        }
+    }
+
+    /**
+     * Inserts a new element at a given position in the array.
+     * The position must be valid, or at the end of the array (idx == size()).
+     */
+    fun insert(idx: Int, data: UByte) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_insert(it, idx, data))
+        }
+    }
+
+    /**
+     * Reverses the order of the elements in the array.
+     */
+    fun invert() {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_invert(it))
+        }
+    }
+
+    /**
+     * Appends a value to the array.
+     */
+    fun pushBack(data: UByte) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_push_back(it, data))
+        }
+    }
+
+    /**
+     * Removes an element from the array by index.
+     */
+    fun remove(idx: Int) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_remove(it, idx))
+        }
+    }
+
+    /**
+     * Sets the size of the array. If the array is grown, reserves elements at the end of the array.
+     * If the array is shrunk, truncates the array to the new size.
+     */
+    fun resize(size: Int) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_resize(it, size))
+        }
+    }
+
+    /**
+     * Changes the Byte at the given index.
+     */
+    operator fun set(idx: Int, data: UByte) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_set(it, idx, data))
+        }
+    }
+
+    /**
+     * Returns the size of the array.
+     */
+    fun size(): Int {
+        return callNative {
+            checkNotNull(Godot.gdnative.godot_pool_byte_array_size(it))
+        }
+    }
+
+    //UTILITIES
+    fun append(byte: Int) {
+        append(byte.toUByte())
+    }
+
+    operator fun set(idx: Int, byte: Int) {
+        this[idx] = byte.toUByte()
+    }
+
+    override fun toString(): String {
+        return "PoolByteArray(${size()})"
+    }
+
+    override fun iterator(): Iterator<UByte> {
+        return IndexedIterator(size(), this::get)
+    }
+
+    internal inline fun <T> callNative(block: MemScope.(CPointer<godot_pool_byte_array>) -> T): T {
+        return memScoped {
+            val ptr = handle.ptr
+            val ret: T = block(ptr)
```

Comment:
Maybe checkNotNull can be called here to avoid duplicates

Reply:
Actually I made a mistake, It's not the result of the lamda that I should checkNotNull but the function we retrieve from Godot.gdnative.
Ex: 
 checkNotNull(Godot.gdnative.godot_X_fun)(args)
not  checkNotNull(Godot.gdnative.godot_X_fun(args))